unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+        initial = RedisClientBase._check_transaction_id_cached_out
+        RedisClientBase._check_transaction_id_cached_out = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase._check_transaction_id_cached_out = initial
+
+    def test_function_check_buffer_message_matching_command_and_id_fixture_call_success(self,
+                                                                                        check_buffer_message_fixture):
+        check_buffer_message_fixture._check_buffer_message_matching_command_and_id()
+        check_buffer_message_fixture._check_buffer_message_matching_command_and_id.assert_called_once()
+
+    def test_function_register_call_success(self, register_fixture):
+        register_fixture.register()
+        register_fixture.register.assert_called_once()
+
+    def test_function_unregister_call_success(self, unregister_fixture):
+        unregister_fixture.unregister()
+        unregister_fixture.unregister.assert_called_once()
+
+    def test_function_on_register_call_success(self, on_register_fixture):
+        on_register_fixture._on_register()
+        on_register_fixture._on_register.assert_called_once()
+
+    def test_function__check_transaction_id_cached_out_call_success(self,
+                                                                    check_transaction_id_cached_out_call_success_fixture):
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out()
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out.assert_called_once()
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self):
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match='The answer message does not contain a valid '
+                                 '\'transaction_id\' member.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+            redis_client_base._check_buffer_message_matching_command_and_id(message)
+
+    def test_register_blocking_true_throws_exception(self):
+        with pytest.raises(RedisAPIException,
+                           match='API registration process timed out. Server will continue processing '
+                                 'your request on the background and will notify you as soon as the '
+                                 'registration has been completed.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=True)
+            redis_client_base.register(is_blocking=True)
+
+    def test_register_blocking_false_published_success(self):
+        # arrange
+        redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+        redis_client_base.register(is_blocking=False)
+        assert redis_client_base._blocking_command_responses["register"]["name"] == area_id
```

Comment:
What are we testing here? When we test a method, we mostly want to test two things (sometimes both):
1. the visible side-effects of that method (i.e. some change on the public variables);
2. the returned output of that method.

So in this case: what happens when `.register()` is called with `is_blocking=False`? In this test, you're checking the value of `redis_client_base._blocking_command_responses`. However, this is a private variable (so we probably don't need to test it). Moreover, we're not testing the other side-effect: we want to be sure that `self.redis_db.publish()` is called with the correct arguments.

I would suggest you to give it a try. The same applies to many other methods in the context of this PR. Please get in touch if you'd like to have more information and share some opinions :)